katsdptelstate/rdb_writer.py

```diff
@@ -97,49 +99,52 @@ def __init__(self, filename):
         self.keys_written = 0
         self.keys_failed = 0
         self._rdb_context = _rdb_file(self.filename)
-        self._fileobj = None
+        self._fileobj = self._rdb_context.__enter__()
 
     def __enter__(self):
-        """Enter RDB writer context, opening RDB file and writing header."""
-        self._fileobj = self._rdb_context.__enter__()
         return self
 
-    def __exit__(self, exc_type, exc_value, traceback):
-        """Exit RDB writer context, closing off file and deleting it if empty."""
+    def __exit__(self, *exc_info):
+        self.close()
+
+    def close(self):
+        """Close off RDB file and delete it if it contains no keys."""
         try:
-            self._rdb_context.__exit__(exc_type, exc_value, traceback)
```

Comment:
Is it still necessary to have the `_rdb_context` construct? You could probably just put the code to close off the file here.

Reply:
Done, thanks for that.